### PR TITLE
Add `llvm_to_defintion` and `definition_to_llvm` conversions for LLVM bindings

### PR DESCRIPTION
**Pull request by Scott-Guest**
Add functions to convert full KORE definitions between the LLVM representation and the Pyk representation

Blocked until runtimeverification/llvm-backend#823 is merged

### CODE REVIEW

File: src/pyk/kllvm/convert.py [hunk 1]
```diff
@@ -2,26 +2,117 @@
...
 
 # -----------
 # pyk -> llvm
 # -----------
 
 
-def kore_to_llvm(pattern: Pattern) -> kllvm.Pattern:
+def _add_attributes(term: Any, attrs: tuple[App, ...]) -> None:
+    for attr in attrs:
+        term.add_attribute(_composite_pattern(attr.symbol, attr.sorts, attr.args))
```
Comment (tothtamas28) [inline src/pyk/kllvm/convert.py:41]: This private helper should follow follow the public functions.

File: src/pyk/kllvm/convert.py [hunk 2]
```diff
@@ -2,26 +2,117 @@
...
+
+
+def kore_definition_to_llvm(definition: Definition) -> kllvm.Definition:
```
Comment (tothtamas28) [inline src/pyk/kllvm/convert.py:44]: Consider names `definition_to_llvm`, `module_to_llvm`, ..., `pattern_to_llvm` (and conversely, `llvm_to_definition`, `llvm_to_module`, ..., `llvm_to_pattern`).

File: src/pyk/kllvm/parser.py [hunk 3]
```diff
@@ -8,16 +8,26 @@
...
-def parse_file(path: str | Path) -> Pattern:
+def _parser_from_path(path: str | Path) -> Parser:
     path = Path(path)
     check_file_path(path)
-    parser = Parser(str(path))
-    return parser.pattern()
+    return Parser(str(path))
+
+
+def parse_pattern_text(text: str) -> Pattern:
```
Comment (tothtamas28) [inline src/pyk/kllvm/parser.py:20]: How about `parse_pattern`, `parse_definition` and `parse_pattern_file`, `parse_definition_file`?

File: src/tests/integration/kllvm/test_convert.py [hunk 4]
```diff
@@ -46,20 +51,59 @@
...
+        'alias-decl',
+        _in_module('alias Foo{S,T}(S) : T where Foo{S,T}(X : S) := X : T [AliasAttr{}()]'),
+    ),
+    ('axiom', _in_module(r'axiom {S} \rewrites{S}(X : S, \dv{SortInt{}}("1")) [AxiomAttr{}()]')),
+    ('claim', _in_module(r'claim {S} \rewrites{S}(X : S, \dv{SortInt{}}("1")) [AxiomAttr{}()]')),
+)
+
+
+@pytest.mark.parametrize('test_id,kore_text', PAT_TEST_DATA, ids=[test_id for test_id, *_ in PAT_TEST_DATA])
+def test_kore_pattern_to_llvm(test_id: str, kore_text: str) -> None:
```
Comment (tothtamas28) [inline src/tests/integration/kllvm/test_convert.py:75]: If you agree with the naming changes suggested above, then:

```suggestion
def test_pattern_to_llvm(test_id: str, kore_text: str) -> None:
```

(and similarly for the rest)

File: src/tests/integration/kllvm/test_convert.py [hunk 5]
```diff
@@ -46,20 +51,59 @@
 )
 
 
-@pytest.mark.parametrize('test_id,kore_text', TEST_DATA, ids=[test_id for test_id, *_ in TEST_DATA])
-def test_kore_to_llvm(test_id: str, kore_text: str) -> None:
+def _in_module(kore_text: str) -> str:
+    return f'[] module FOO {kore_text} endmodule []'
```
Comment (tothtamas28) [inline src/tests/integration/kllvm/test_convert.py:55]: This can go before `PAT_TEST_DATA`.

File: src/tests/integration/kllvm/test_parser.py [hunk 6]
```diff
@@ -3,31 +3,67 @@
...
+
+def test_parse_definition_file(tmp_path: Path) -> None:
+    # Given
+    kore_text = """[]
+
+module FOO
+  axiom {S}A{}(B{}(),C{}()) [group{}("foo")]
+endmodule
+[concrete{}()]
+"""
```
Comment (tothtamas28) [inline src/tests/integration/kllvm/test_parser.py:44]: These are quite ugly, consider reading the content from file.

**Comment by Scott-Guest** [comment]
@tothtamas28 Oops I let this PR fall to the wayside, addressed all your comments

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**